src/control/zoom/ZoomControl.cpp

```diff
@@ -18,7 +18,11 @@ auto onScrolledwindowMainScrollEvent(GtkWidget* widget, GdkEventScroll* event, Z
                 (event->direction == GDK_SCROLL_UP || (event->direction == GDK_SCROLL_SMOOTH && event->delta_y < 0)) ?
                         ZOOM_IN :
                         ZOOM_OUT;
-        zoom->zoomScroll(direction, {event->x, event->y});
+        // use screen pixel coordinates for the zoom center
+        // as relative coordinates depend on the changing zoom level
+        int rx, ry;
+        gdk_window_get_root_coords(gtk_widget_get_window(widget), 0, 0, &rx, &ry);
```

Comment:
Consider using [`gdk_window_get_root_origin`](https://docs.gtk.org/gdk3/method.Window.get_root_origin.html) here and/or making a wrapper utility method (`Util::toScreenCoords(WindowWidget, Point) -> Point`).